addons/stock/models/stock_move.py

```diff
@@ -1515,9 +1515,9 @@ def _action_done(self, cancel_backorder=False):
             # To know whether we need to create a backorder or not, round to the general product's
             # decimal precision and not the product's UOM.
             rounding = self.env['decimal.precision'].precision_get('Product Unit of Measure')
-            if float_compare(move.quantity_done, move.product_uom_qty, precision_digits=rounding) < 0:
+            if float_compare(move.quantity_done, move.product_uom_qty, precision_digits=rounding) < 0 or self.env.context.get('bo_production_id'):
```

Comment:
should try to remove context. replace by a function that just return true in stock and which is override in mrp